test/logflare_web/controllers/api/source_controller_test.exs

```diff
@@ -0,0 +1,45 @@
+defmodule LogflareWeb.Api.SourceControllerTest do
+  use ExUnit.Case
+  use LogflareWeb.ConnCase
+
+  import Logflare.Factory
+  import Assertions
+
+  alias Logflare.Sources.Counters
+  alias Logflare.Teams
+
+  setup do
+    insert(:plan, name: "Free")
+    u1 = insert(:user)
+    u2 = insert(:user)
+    Teams.create_team(u1, %{name: "u1 team"})
+    Teams.create_team(u2, %{name: "u2 team"})
+
+    s1 = insert(:source, public_token: Faker.String.base64(16), user_id: u1.id)
+    s2 = insert(:source, user_id: u1.id)
+    s3 = insert(:source, user_id: u2.id)
+
```

Comment:
we can cut this down to 
```
user = insert(:user)
team = insert(:team, user: user)
source = insert(:source, user: user)
```
Might already have a team factory, not sure.

Also, we don't need to start the Counters, we can mock its behaviour instead (we're going to refactor it heavily)

Reply:
can also just pass  at the end of setup

```
[user: user, team: team, source: source]
```
since the api test only needs to test that the api works, don't need to test logic for fetching of multiple sources

Reply:
will change, this was a big copy-paste from the previous code and I wanted to avoid messing too much with it but you are right it needs a cleanup to standardize test approaches

As for Counters the issue is actually the lack of an ETS table that is initialized with the process itself. We do not have an easy way now to mock it from what I can see (no way to define the implementation to be used).